tests/components/random/test_config_flow.py

```diff
@@ -0,0 +1,201 @@
+
+
+@pytest.mark.parametrize(
+    (
+        "entity_type",
+        "extra_input",
+        "extra_options",
+    ),
+    (
+        (
+            "binary_sensor",
+            {},
+            {},
+        ),
+        (
+            "sensor",
+            {
+                "device_class": SensorDeviceClass.POWER,
+                "unit_of_measurement": UnitOfPower.WATT,
+            },
+            {
+                "device_class": SensorDeviceClass.POWER,
+                "unit_of_measurement": UnitOfPower.WATT,
+                "minimum": 0,
+                "maximum": 20,
+            },
+        ),
+        (
+            "sensor",
+            {},
+            {"minimum": 0, "maximum": 20},
+        ),
+    ),
+)
+async def test_config_flow(
+    hass: HomeAssistant,
+    entity_type: str,
+    extra_input: dict[str, Any],
+    extra_options: dict[str, Any],
+) -> None:
+    """Test the config flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.MENU
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {"next_step_id": entity_type},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == entity_type
+
+    with patch(
+        "homeassistant.components.random.async_setup_entry", wraps=async_setup_entry
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "name": "My random entity",
+                **extra_input,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "My random entity"
+    assert result["data"] == {}
+    assert result["options"] == {
+        "name": "My random entity",
+        "entity_type": entity_type,
+        **extra_options,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.parametrize(
+    ("device_class", "unit_of_measurement"),
+    [
+        (SensorDeviceClass.POWER, UnitOfEnergy.WATT_HOUR),
+        (SensorDeviceClass.ILLUMINANCE, UnitOfEnergy.WATT_HOUR),
+    ],
+)
+async def test_wrong_uom(
+    hass: HomeAssistant, device_class: SensorDeviceClass, unit_of_measurement: str
+) -> None:
+    """Test entering a wrong unit of measurement."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.MENU
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {"next_step_id": "sensor"},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "sensor"
+
+    with pytest.raises(Invalid):
```

Comment:
```suggestion
    with pytest.raises(Invalid, match="todo"):
```

We should match the error message to be sure the correct one is raised